src/dc_msg.rs

```diff
@@ -31,7 +32,7 @@ pub struct dc_msg_t<'a> {
     pub timestamp_sort: i64,
     pub timestamp_sent: i64,
     pub timestamp_rcvd: i64,
-    pub text: *mut libc::c_char,
+    pub text: Option<String>,
```

Comment:
I'd kind of argue that this should be just be `String` and use `""` if there is nothing.  I think the `None` case is really an artefact of the C API and only needs to be replicated on that boundary layer.  A lot of things probably become easier if you can just assume this is always a valid string instead of having to handle the Option.

I don't feel too strongly though, this PR is certainly already an improvement as is!